Java/Algorithms/Moore_Algorithm.java

```diff
@@ -0,0 +1,43 @@
+import java.util.Scanner;
```

Comment:
Add a Multi-Line Comments explaining the Algorithm in a few lines. Nothing fancy but it will help people to understand the main pitch of the algorithm.